### PR TITLE
[Site] migrate to Docusaurus

### PR DESCRIPTION
**Pull request by yangshun**
## Summary

This website is due for an upgrade. I migrated it to [Docusaurus 2](https://docusaurus.io) with the approval of @zhuhanming.

Benefits:

- Dark mode
- Single page app
- Better content reading experience with table of contents
- More out-of-the-box features
- More powerful templating (via MDX which allows you to write JSX)

I tried to preserve the history as much as possible, especially for the coursework writeups.

We'll need to add the `gh-pages` branch and configure the hosting later as GitHub doesn't automatically host non-Jekyll sites.

## Test Plan

Try it out at https://cs3216.vercel.app/

Things to check:

- Coursework pages
- Student pages
- About page

### CODE REVIEW

**Comment by zhuhanming** [comment]
Looks great! I'll take a closer look tonight (when I get home), but I noticed that the final project page dies when I tried it on Safari (though it works on Chrome). The other pages work fine.

The console error message is:

```
[Error] RangeError: date value is not finite in DateTimeFormat format()
format
[native code]
m — 8101345c.7bafd357.js:1:5102
ai — main.7792bd9d.js:2:197465
Hl — main.7792bd9d.js:2:249705
_u — main.7792bd9d.js:2:236911
Tu — main.7792bd9d.js:2:236839
xu — main.7792bd9d.js:2:236702
gu — main.7792bd9d.js:2:233689
gu
(anonymous function) — main.7792bd9d.js:2:183360
(anonymous function) — main.7792bd9d.js:2:287135
Va — main.7792bd9d.js:2:183306
Ha — main.7792bd9d.js:2:183241
du — main.7792bd9d.js:2:231066
enqueueSetState — main.7792bd9d.js:2:187215
(anonymous function) — main.7792bd9d.js:2:278682
(anonymous function) — main.7792bd9d.js:2:13851
promiseReactionJob

	ll (main.7792bd9d.js:2:220720)
	(anonymous function) (main.7792bd9d.js:2:222819)
	fo (main.7792bd9d.js:2:186853)
	ml (main.7792bd9d.js:2:224555)
	Pu (main.7792bd9d.js:2:241064)
	Pu
	(anonymous function) (main.7792bd9d.js:2:287135)
	Au (main.7792bd9d.js:2:237803)
	gu (main.7792bd9d.js:2:233914)
	gu
	(anonymous function) (main.7792bd9d.js:2:183360)
	(anonymous function) (main.7792bd9d.js:2:287135)
	Va (main.7792bd9d.js:2:183306)
	Ha (main.7792bd9d.js:2:183241)
	du (main.7792bd9d.js:2:231066)
	enqueueSetState (main.7792bd9d.js:2:187215)
	(anonymous function) (main.7792bd9d.js:2:278682)
	(anonymous function) (main.7792bd9d.js:2:13851)
	promiseReactionJob
```

**Comment by zhuhanming** [comment]
Also, I noticed that the batch photos for 2019 and 2020 seem to be missing:

```
[Error] Failed to load resource: the server responded with a status of 404 () (2020.jpg, line 0)
https://cs3216.vercel.app/img/class-photos/2020.jpg
```

**Comment by yangshun** [comment]
> the final project page dies when I tried it on Safari (though it works on Chrome). The other pages work fine.

Ah I found a seemingly invalid date string in the `assignments.json`

> Also, I noticed that the batch photos for 2019 and 2020 seem to be missing:

Great catch! This is due to the images for 2019 and 2020 to use `.JPG` extension. Have renamed them to `.jpg`.

Thanks for the thorough review Hanming!

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2014 NUS Modifications
+Copyright (c) 2014 Cs3216
```
Comment (zhuhanming) [inline LICENSE]: Nit: Shall we capitalise to `CS3216`?

File: website/coursework/assignments.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "proposal_submission": "2021-09-27T23:59:00",
+    "progress_report_1_submission": "2021-10-11T23:59:00",
+    "progress_report_2_submission": "2021-10-25T23:59:00",
+    "in_class_progress_report": "2021-11-01T18:30:00",
+    "poster_session": "2021-11-10",
+    "is_poster_session_date_tentative": true,
+    "final_report_submission": "2021-11-21T23:59:00",
+    "google_sheet": "https://docs.google.com/spreadsheets/d/1J-g4Yiq-7_-og09ifzYk_KVHWBICcIs8mmuyTFt-PYQ/edit?usp=sharing"
+  }
+}
```
Comment (zhuhanming) [inline website/coursework/assignments.json]: nit: new line below

File: website/coursework/product-design/index.md [hunk 3]
```diff
@@ -103,12 +122,15 @@ List the above in a _user persona_ card. A persona is a fictional representation
 
 ![Example Persona](img/persona.jpg)
 
-<small>Source: <https://uxdesign.cc/meetup-a-usability-case-study-e909c33f1e3e></small>
+<small>Source: https://uxdesign.cc/meetup-a-usability-case-study-e909c33f1e3e</small>
```
Comment (zhuhanming) [inline website/coursework/product-design/index.md:125]: Hmm should we make these links clickable? I think both are fine, so long we standardise.

File: website/coursework/product-design/index.md [hunk 4]
```diff
@@ -246,23 +283,29 @@ After you have completed the testing, it is now time to review the results and i
...
+If you feel that you are not ready for this stage yet, you may perform more rounds of design iteration and user testing with wireframes. Be sure to include the results and name the folders appropriately.
+
+:::
 
 Compared to a wireframe, a mockup will usually look more similar to how the final app will turn out. Colours, fonts, logos, images and user interface elements from the target platform are used in place of wire boxes. However, like wireframes, mockups remain static and will require manual transitioning when doing user testing with them.
 
 ![Example Mobile Prototype](img/mobile-design-mockup.png)
 
-<small>Source: <https://www.nngroup.com/articles/wireflows/></small>
+<small>Source: https://www.nngroup.com/articles/wireflows/</small>
```
Comment (zhuhanming) [inline website/coursework/product-design/index.md:308]: Ditto

File: website/coursework/product-design/index.md [hunk 5]
```diff
@@ -272,13 +315,15 @@ If you are planning to work on a prototype, there is no need to submit the desig
 
 ![Example Travel Design Prototype](img/travel-design-mockup.png)
 
-<small>Source: <http://www.tifftam.co/goji/></small>
+<small>Source: http://www.tifftam.co/goji/</small>
```
Comment (zhuhanming) [inline website/coursework/product-design/index.md:318]: Ditto

File: website/coursework/product-design/index.md [hunk 6]
```diff
@@ -165,31 +198,31 @@ Other good resources for design inspiration include [Dribbble](https://dribbble.
...
 
-<iframe width="736" height="414" src="https://www.youtube.com/embed/JMjozqJS44M?start=124" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
+<iframe width="736" height="414" src="https://www.youtube.com/embed/JMjozqJS44M?start=124" frameBorder="0" allow="autoplay; encrypted-media" allowFullScreen></iframe>
 
 Wireframes are ["low-fidelity" designs](https://www.nngroup.com/articles/ux-prototype-hi-lo-fidelity/). This means that they should _not_ contain any design-specific elements such as colours, font or unnecessary details like user-generated content and images. These distract from the aspects that are more important at this current stage of development, and slow down the speed at which they are produced. Instead, focus on the relative size and position of elements, key user interface elements, and interactions such as which elements can be clicked or tapped, dragged, and so on.
 
 ![Example Mobile Wireframe](img/wireframe-mobile.jpg)
 
-<small>Source: <https://www.flickr.com/photos/anthonyarmendariz/4448219885/in/pool-1070674@N20/></small>
+<small>Source: https://www.flickr.com/photos/anthonyarmendariz/4448219885/in/pool-1070674@N20/</small>
```
Comment (zhuhanming) [inline website/coursework/product-design/index.md:211]: Ditto

File: website/coursework/product-design/index.md [hunk 7]
```diff
@@ -165,31 +198,31 @@ Other good resources for design inspiration include [Dribbble](https://dribbble.
...
 
 This set of wireframes will be shown to the user during testing. As such, keep a copy without the links between buttons and screens and without more than one screen showing at once - after all, users can't do that in the actual app! (You may add annotations or comments for your own reference, but remember to hide them before starting user testing.)
 
 ![Example Web Wireframe](img/wireframe-web.jpg)
 
-<small>Source: <https://www.flickr.com/photos/mockupbuilder/12987528713/in/pool-ilovewireframes/></small>
+<small>Source: https://www.flickr.com/photos/mockupbuilder/12987528713/in/pool-ilovewireframes/</small>
```
Comment (zhuhanming) [inline website/coursework/product-design/index.md:219]: Ditto

File: website/package.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      "not dead",
+      "not op_mini all"
+    ],
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  }
+}
```
Comment (zhuhanming) [inline website/package.json]: nit: new line

File: website/students/StudentsList.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+          display: hideImage ? 'none' : 'inherit',
+        }}
+      />
+      <div className="row">
+        {students
+          .filter((batch) => batch.year === year)[0]
+          .students.map(({name, id, faculty, blog, one_liner}) => (
+            <div className="col col--4 margin-bottom--lg" key={name}>
+              <div
+                className="card shadow--tl padding-top--lg text--center"
```
Comment (zhuhanming) [inline website/students/StudentsList.js]: Actually, I'm thinking about the shadow - I personally feel that the drop shadow here is slightly incongruent with how the rest of the website feels, though I do understand that it's one way of showing the border of the cards. Would like to hear your thoughts on this.

**Review by zhuhanming — APPROVED** [review]
LGTM! Mostly minor nits, thanks for helping to revamp the website! 👍🏼 

Btw, has the vercel app been redeployed? It's still broken on Safari for me.

File: LICENSE (hunk 1 above)
Comment (yangshun) [reply LICENSE]: Absolutely, that's a mistake on my part

File: website/coursework/product-design/index.md (hunk 3 above)
Comment (yangshun) [reply website/coursework/product-design/index.md:125]: Yes. Lemme do it in a followup PR where I'll use more semantic ways

File: website/package.json (hunk 8 above)
Comment (yangshun) [reply website/package.json]: Somehow my editor stopped adding newline at the end of file on save. Added it back!

File: website/students/StudentsList.js (hunk 9 above)
Comment (yangshun) [reply website/students/StudentsList.js]: I agree. I'll change to a 1px grayish border

**Comment by yangshun** [comment]
> Btw, has the vercel app been redeployed? It's still broken on Safari for me.

I just tested it on my personal MBP and it's fine. Maybe you're viewing a cached version

**Comment by zhuhanming** [comment]
> I just tested it on my personal MBP and it's fine. Maybe you're viewing a cached version

Ah yes, it works for me now! Thanks!

**Comment by zhuhanming** [comment]
Btw any action required on my end with regards to the deployment of the new site? In the meantime, I'll tell the students to use https://cs3216.vercel.app/ while the main site is down.

**Pull request merged**